src/workerd/api/streams/queue.c++

```diff
@@ -647,10 +638,9 @@ void ByteQueue::handlePush(
 
           // We have completely consumed the data in this entry and can safely free
           // our reference to it now. Yay!
-          auto released = kj::mv(next);
-          state.buffer.pop_front();
+          auto dropped = KJ_ASSERT_NONNULL(state.buffer.pop());
```

Comment:
It's worth considering that replacing `pop_front()` with `auto dropped = buffer.pop()` isn't exactly equivalent: the former will destroy the object immediately, the latter will destroy the object at the end of the lexical scope. I didn't see anywhere that would cause an actual problem, but haven't been looking too closely.

Reply:
Yep, I evaluated carefully for this and this is actually intentional. If someone did happen to do a `peek()` call before calling `pop()` and still happens to have a reference in the current lexical scope, this should at least make it less likely for that to become invalid in that current scope. Obviously we're not preventing all problems just possibly some.